packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -88,13 +92,17 @@ export class ContextualMenuBase extends BaseComponent<IContextualMenuProps, ICon
   private _enterTimerId: number | undefined;
   private _targetWindow: Window;
   private _target: Element | MouseEvent | IPoint | null;
-  private _classNames: IContextualMenuClassNames;
   private _isScrollIdle: boolean;
   private _scrollIdleTimeoutId: number | undefined;
   private _processingExpandCollapseKeyOnly: boolean;
 
   private _adjustedFocusZoneProps: IFocusZoneProps;
 
+  private _classNames = getClassNames(this.props.styles, {
```

Comment:
Are you sure this is the right place to call this? Styles is a function taking in `IContextMenuStyleProps`, which consumers could decide to provide with dynamic output. I feel like this should be called in render and its output can't assumed to be static.

Reply:
This was a pattern suggested by I believe @dzearing, but when @bennettclark was working on Facepile.  It's a solution to solve calling `getClassNames()` for each function it's used in (3 in this file including the render).  My understanding was a private method won't work if style props are set from local variables (like `disabled` based on something in the item function), but if like this case it just has `styles` and `theme` that a private method is preferred.  If this conflicts with something an app dev might do while customizing styles please let me know.

Reply:
I wasn't talkiing about having it declared here as a class member, but rather the need to have it initialized here. It's only used in render functions, so it doesn't have to be initialized here as long as it's set at the top of render(). I believe that has been the standard as props will change on render, allowing getClassNames to be called with the new props values on each render call.

Reply:
but its also needed in `_renderHeaderMenuItem()` and `_renderSectionItem()`.

Reply:
Which are called from `render`. If the class member is set at the top of render, it will be set for those functions.